### DISCUSSION TITLE
Monitor Outgoing HTTP

### DISCUSSION THREAD

**Discussion by matrenner** [javamelody/javamelody#1002] [Q&A] [answered]
Hello,

Is there a way to setup JavaMelody to collect statistics on outgoing HTTP calls? For stats that would make sense like mean time, max time, hits etc.

Thanks

**Comment by evernat** [comment, marked as answer]
Hello Mat,

If you use the javamelody spring boot starter and if you use a Spring RestTemplate defined as a bean in your configuration, then yes the stats of the outgoing http calls are displayed in the Spring's stats.
Or if you use Spring and you import `net/bull/javamelody/monitoring-spring.xml` in Spring's context and if you use a Spring RestTemplate defined as bean in your Spring's context, then yes again.

Otherwise, no.
But you can still monitor a "service" method around the http calls [using Spring](https://github.com/javamelody/javamelody/wiki/UserGuide#9-business-facades-if-spring) or [using java interfaces](https://github.com/javamelody/javamelody/wiki/UserGuideAdvanced#business-facades-without-ejb3-and-without-spring-and-without-guice) or you can just monitor a piece of code, for example:

```java
import net.bull.javamelody;

    try (Stopwatch stopwatch = new Stopwatch("name you want")) {
        // your piece of code
    }
```

**Reply by matrenner** [reply]
Great! Thank you. That is very helpful
